dessia_common/document_generator.py

```diff
@@ -48,46 +107,76 @@ def add_page_breaks(self, num_page_breaks: int):
     def add_table(self, rows: List[List[str]]) -> 'DocxWriter':
         """ Add table to the document. """
         document = self.document
-        table = document.add_table(rows=1, cols=len(rows[0]))
+        table = document.add_table(rows=1, cols=len(rows[0]), style="TableGrid")
         for i, cell in enumerate(table.rows[0].cells):
             cell.text = rows[0][i]
         for row in rows[1:]:
             row_cells = table.add_row().cells
             for i, cell in enumerate(row_cells):
                 cell.text = row[i]
-        writer = DocxWriter(self.filename)
-        writer.document = document
-        writer.headings = self.headings
-        return writer
+        self.document = document
+        return self
 
-    def add_header_footer(self, text: str, is_header: bool = True, align: str = 'center') -> 'DocxWriter':
+    def add_header_footer(self, is_header: bool = True) -> 'DocxWriter':
         """ Add a header or footer to the document. """
+        section = self.document.sections[-1]
+        if is_header and self.header is not None:
```

Comment:
There is something wrong about the way the mutualization of header/footer is done. Maybe you need a concept that links Header/Footer and its section.

By using OO programming, you should not need to test if is_header everywhere.

Reply:
@GhislainJ 
I added the class Section...